subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -12,11 +12,46 @@
 
 LOG_MODULE_REGISTER(dfu_target_mcuboot, CONFIG_DFU_TARGET_LOG_LEVEL);
 
+#define MAX_FILE_SEARCH_LEN 500
 #define MCUBOOT_HEADER_MAGIC 0x96f3b83d
 
 static struct flash_img_context flash_img;
 static size_t offset;
 
+int dfu_ctx_mcuboot_set_b1_file(char *file, bool s0_active, char **update)
+{
+	if (file == NULL   ||  update == NULL) {
+		return -EINVAL;
+	}
+
+	/* Ensure that 'file' is null-terminated. */
+	if (strnlen(file, MAX_FILE_SEARCH_LEN) == MAX_FILE_SEARCH_LEN) {
+		return -EINVAL;
+	}
+
+	/* We have verified that there is a null-terminator, so this is safe */
+	char *space = strstr(file, " ");
+
+	if (space == NULL) {
+		/* Could not find space separator in input */
+		*update = NULL;
+
+		return 0;
```

Comment:
Why not an error?

Reply:
This is not an error situation. If you see how it is used, this function is also used to check whether or not there is a space separated file.

It is indicated in the doc of the function that you check if `update` is `NULL` to see if there was space separator in the input file.